fingerprint2.js

```diff
@@ -209,6 +216,94 @@
       }
       return keys
     },
+    isFlashKey: function (keys){
+      if (!this.options.excludeIsFlash){
+          keys.addPreprocessedComponent({key: 'is_flash', value:this.getIsFlash()})
+      }
+      return keys
+    },
+    getIsFlash: function(){
+      var objPlugin = navigator.plugins["Shockwave Flash"];
+      if (objPlugin) {
+        return true;
+      }
+      return false;
+    },
+    flashVersionKey: function(keys){
+       if (!this.options.excludeFlashVersion){
+          keys.addPreprocessedComponent({key: 'flash_version', value:this.getFlashVersion()})
+      }
+      return keys
+    },
+    getFlashVersion: function() {
+      if (this.getIsFlash()) {
+        var objPlayerVersion = swfobject.getFlashPlayerVersion();
+        return objPlayerVersion.major + "." + objPlayerVersion.minor + "." + objPlayerVersion.release;
+      }
+      return "unknown";
+    },
+    isSilverlightKey: function(keys){
+      if (!this.options.excludeIsSilverlight){
+          keys.addPreprocessedComponent({key: 'is_silverlight', value:this.getIsSilverlight()})
+      }
+      return keys
+    },
+    getIsSilverlight: function() {
+      var objPlugin = navigator.plugins["Silverlight Plug-In"];
+      if (objPlugin) {
+        return true;
+      }
+      return false;
+    },
+    silverlightVersionKey: function(keys){
+      if (!this.options.excludeSilverlightVersion){
+          keys.addPreprocessedComponent({key: 'silverlight_version', value:this.getSilverlightVersion()})
+      }
+      return keys
+    },
+    getSilverlightVersion: function() {
+      if (this.getIsSilverlight()) {
+        var objPlugin = navigator.plugins["Silverlight Plug-In"];
+        return objPlugin.description;
+      }
+      return "unknown";
+    },
+    isMimeTypesKey: function(keys){
```

Comment:
This is redundant with the `mimeTypesKey` component, isn't it? It's the length of that other component. Does the duplication make sense?

Reply:
May be "isMimeTypeKey"  no good. 
Remove this in 73f642d.